Sources/CSwiftVideo/shim.cpp

```diff
@@ -270,6 +275,92 @@ extern "C" {
         return 1;
     }
 
+    static int vp9_bitdepth_colorspace_sampling(ExpGolomb& decoder, VP9FrameProperties* props) {
+        props->bitDepth = 8;
+        if(props->profile >= 2) {
+            props->bitDepth = *decoder.get_bits(1) ? 12 : 10;
+        }
+        props->colorSpace = static_cast<VP9ColorSpace>(*decoder.get_bits(3));
+        if(props->colorSpace != VP9ColorSpace_SRGB) {
+            props->fullSwingColor = *decoder.get_bits(1); // movie = 0, full = 1
+            if(props->profile == 1 || props->profile == 3) {
+                props->subSamplingX = *decoder.get_bits(1);
+                props->subSamplingY = *decoder.get_bits(1);
+                decoder.get_bits(1); // reserved 0
+            } else {
+                props->subSamplingX = props->subSamplingY = 1;
+            }
+        } else {
+            props->subSamplingX = props->subSamplingY = 0;
+            decoder.get_bits(1); // reserved 0
+            if(props->profile != 1 && props->profile != 3) {
+                return 0;
+            }
+        }
+        return 1;
+    }
+    static int vp9_frame_size(ExpGolomb& decoder, VP9FrameProperties* props) {
+        props->width = *decoder.get_bits(16) + 1;
+        props->height = *decoder.get_bits(16) + 1;
+        auto has_scaling = decoder.get_bits(1);
+        if(has_scaling && *has_scaling) {
+            props->displayWidth = *decoder.get_bits(16) + 1;
+            props->displayHeight = *decoder.get_bits(16) + 1;
+        } else {
+            props->displayWidth = props->width;
+            props->displayHeight = props->height;
+        }
+        return 1;
+    }
+    int vp9_is_keyframe(const void* data, int64_t size, int* is_keyframe) {
+        auto decoder = ExpGolomb((uint8_t*)data, size);
+        if(*decoder.get_bits(2) != 0b10) {
+            return 0;
+        }
+        auto version = *decoder.get_bits(1);
+        auto high = *decoder.get_bits(1);
+        auto profile = (high << 1) + version;
+        if(profile == 3) {
+            decoder.get_bits(1); // reserved
+        }
+        if(*decoder.get_bits(1)) { // show_existing_frame - not a new frame
+            return 0;
+        }
+        *is_keyframe = !(*decoder.get_bits(1));
```

Comment:
May I ask a question?  old h264 we can use `kCMSampleAttachmentKey_DependsOnOthers ` to determine is keyframe or not, why here is not? sry I don't master the vp9 formate now. https://github.com/jgh-/VideoCore-Inactive/blob/859dc77dbe29a6d052beda92f2d7f407dbd303e7/transforms/Apple/H264Encode.mm#L66